### PR TITLE
Chore: Rewrite ldap login test to standard library

### PR DESCRIPTION
**Pull request by ceh**
**What this PR does / why we need it**:

Rewrite ldap login test to standard library.

**Which issue(s) this PR fixes**:

Part of #22844.

### CODE REVIEW

File: pkg/login/ldap_login_test.go [hunk 1]
```diff
@@ -8,65 +8,41 @@ import (
...
 
 var errTest = errors.New("test error")
 
-func TestLDAPLogin(t *testing.T) {
-	Convey("Login using ldap", t, func() {
-		Convey("Given ldap enabled and no server configured", func() {
-			setting.LDAPEnabled = true
+func TestLoginUsingLDAP(t *testing.T) {
+	LDAPLoginScenario(t, "When LDAP enabled and no server configured", func(sc *LDAPLoginScenarioContext) {
+		setting.LDAPEnabled = true
```
Comment (aknuds1) [inline pkg/login/ldap_login_test.go:19]: Not entirely within the scope of the PR, but we should fix this while we're cleaning up:
```suggestion
		origLDAPEnabled = setting.LDAPEnabled
		t.Cleanup(func() {
			setting.LDAPEnabled = origLDAPEnabled
		})
		setting.LDAPEnabled = true
```

File: pkg/login/ldap_login_test.go [hunk 2]
```diff
@@ -8,65 +8,41 @@ import (
...
-				})
+		assert.True(t, enabled)
+		assert.True(t, sc.LDAPAuthenticatorMock.loginCalled)
+	})
 
-				Convey("it should return false", func() {
-					So(enabled, ShouldBeFalse)
-				})
+	LDAPLoginScenario(t, "When LDAP disabled", func(sc *LDAPLoginScenarioContext) {
+		setting.LDAPEnabled = false
```
Comment (aknuds1) [inline pkg/login/ldap_login_test.go:38]: Not entirely within the scope of the PR, but we should fix this while we're cleaning up:
```suggestion
		origLDAPEnabled = setting.LDAPEnabled
		t.Cleanup(func() {
			setting.LDAPEnabled = origLDAPEnabled
		})
		setting.LDAPEnabled = true
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Generally looks good, but please see suggestions for cleanup.

File: pkg/login/ldap_login_test.go (hunk 1 above)
Comment (ceh) [reply pkg/login/ldap_login_test.go:19]: Good catch! I fixed it in https://github.com/grafana/grafana/pull/29998/commits/f88a3a43b287011a7869eb8d13515df169bc66c0 by preserving the original value within the `LDAPLoginScenario` function.

**Review by aknuds1 — APPROVED** [review]

**Comment by aknuds1** [comment]
Thanks!

**Pull request merged**